joomla/joomla-cms
# [5.0] Remove the last code artefacts for Microsoft SQL Server / SQL Azure databases

Pull Request for Issue # .

### Summary of Changes

This pull request (PR) removes the last code artefacts for using Joomla with Microsoft SQL Server / SQL Azure databases.

These database types are not supported anymore since Joomla 4.0, but there is the "SqlsrvChangeItem" class left in the "Joomla\CMS\Schema\ChangeItem" namespace, which is of no use in J4 and should be removed with 5.0.0.

For deprecation see PR #38428 .

This PR removes that class and corresponding code:
- Remove file "SqlsrvChangeItem.php".
- Remove alias for that file in "classmap.php".
- Remove obsolete check for server type "mssql" in file "ChangeItem.php".

In addition, this PR removes an obsolete special handling for server type "mssql" in file "ChangeSet.php". The "sqlazure" subfolder doesn't exist anymore for core update SQL scripts since Joomla 4.0, and 3rd party extensions (which the database checker also checks since 4.0) are not expected to have any update SQL scripts for Microsoft SQL Server / SQL Azure.

All changes affect only the database checker ("System -> Maintenance -> Database", i.e. `administrator/index.php?option=com_installer&view=database`).

### Additional information

The unit tests for the "Joomla\CMS\Schema\ChangeItem" namespace have never been migrated from Joomla 3 to Joomla 4, so in J4 we don't have any unit tests for the subject ot this PR.

I am preparing a PR to add unit tests for that namespace for the 4.2-dev branch (or maybe 4.3-dev), but that will still take some time until completed.

### Testing Instructions

Code review, or make a new installation with the package built for this PR by drone and verify that the database checker works.

### Actual result BEFORE applying this Pull Request

Obsolete code for Microsoft SQL Server / SQL Azure databases in the "Joomla\CMS\Schema\ChangeItem" namespace.

### Expected result AFTER applying this Pull Request

No obsolete code for Microsoft SQL Server / SQL Azure databases in the "Joomla\CMS\Schema\ChangeItem" namespace.

The database checker works as well as without this PR.

### Documentation Changes Required

None.
